packages/apollo-server-lambda/src/ApolloServer.ts

```diff
@@ -41,6 +39,49 @@ export class FileUploadRequest extends Readable {
   headers!: IncomingHttpHeaders;
 }
 
+// Lambda has two ways of defining a handler: as an async Promise-returning
+// function, and as a callback-invoking function.
+// https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html The async
+// variety was introduced with Lambda's Node 8 runtime. Apparently the
+// callback-invoking variety was removed with their Node 14 runtime (their docs
+// don't mention this anywhere but our users have reported this:
+// https://github.com/apollographql/apollo-server/issues/1989#issuecomment-778982945).
+// While AWS doesn't directly support pre-Node-8 runtimes any more, it's
+// possible some users are using a Custom Runtime that still requires the Node 6
+// version, and Apollo Server still technically supports Node 6. So for now, we
+// define an async handler and use this function to convert it to a function
+// that can work either as an async or callback handler.
+//
+// (Apollo Server 3 will drop Node 6 support, at which point we should just make
+// this package always return an async handler.)
```

Comment:
I've been `// AS3`-ing these sorts of things in the source, though I suppose we will also find `Apollo Server 3` if we look. 😉

Reply:
I've just been heavily filing tickets on https://github.com/apollographql/apollo-server/milestone/16 :)